pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -137,7 +138,10 @@ def store_package_for_mirroring(repo, pkgid, location_href):
         location_href: The relative path to the package within the repository
     """
     global pkgid_to_location_href
-    pkgid_to_location_href[str(repo.pk)][pkgid] = location_href
+    # this shouldn't really add the location_href to a list, really it ought to set the value
+    # but unfortunately some repositories have the same packages present in multiple places
+    # same pkgid, >1 different location_hrefs
+    pkgid_to_location_href[str(repo.pk)][pkgid].append(location_href)
```

Comment:
you mentioned ` Multiple package entries with the same NEVRA, same build time, and (probably) same checksum, with the same or different filename / location_href, are present in the metadata`
does it make sense to have it a set() instead in case there will be 1+ same location_href?

Reply:
Done